### PR TITLE
feat: [IDP-214]: Handle secret deletion

### PR DESCRIPTION
**Pull request by vikyathharekal**
## Describe your changes
1. Secret delete event -> Update backstage env variable record in DB with isDeleted = true.
2. In the plugin or configurations page, isDeleted = true will be returned in the API. UI can show an ! mark.
3. Once a new secret is selected in UI on the plugin or configurations page, UI should update isDeleted = false. backstage env variable in the DB will be marked as isDeleted = false.
4. This is backward compatible as we are introducing a new field which is by default false.
5. Observed issue in mapping on the spring data layer as it wasn't understanding the new isDeleted field. Added a migration to add the isDeleted field to all the records.

## Checklist
- [x] I've documented the changes in the PR description.
- [x] I've tested this change either in PR or local environment.
- [x] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- 
  Specific builds
  - `trigger manager`
  - `trigger dms`
  - `trigger ng_manager`
  - `trigger cvng `
  - `trigger cmdlib`
  - `trigger template_svc`
  - `trigger events_fmwrk_monitor`
  - `trigger event_server`
  - `trigger change_data_capture`
  -  Trigger multiple builds together: For eg: `trigger dms, manager`
- Immutable delegate `trigger publish-delegate`
</details>

## [Latest PR Check Triggers](https://github.com/harness/harness-core/blob/develop/.github/pull_request_template.md)

<details>
  <summary>PR Check triggers</summary>
You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ut0, ut1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
  - CodeFormat: `trigger codeformat`
  - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- Trigger CommonChecks: `trigger commonchecks`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- UnitTests-ALL: `trigger uts`
- UnitTests-0: `trigger ut0`
- UnitTests-1: `trigger ut1`
- UnitTests-2: `trigger ut2`
- UnitTests-3: `trigger ut3`
- UnitTests-4: `trigger ut4`
- UnitTests-5: `trigger ut5`
- UnitTests-6: `trigger ut6`
- UnitTests-7: `trigger ut7`
- UnitTests-8: `trigger ut8`
- UnitTests-9: `trigger ut9`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
- Go Build: `trigger gobuild`
- Validate_Reviews: `trigger review`
- Module Dependency Check: `trigger mdc`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/50844)

### CODE REVIEW

File: idp-service/contracts/openapi/v1/openapi.yaml [hunk 1]
```diff
@@ -2350,8 +2355,12 @@ components:
 tags:
   - name: AccountInfo
     description: Account Information
+  - name: AllowList
```
Comment (vikyathharekal) [inline idp-service/contracts/openapi/v1/openapi.yaml:2358]: Just sorted alphabetically

**Review by sathish-soundarapandian — DISMISSED** [review]

**Review by sandeepa-kv-jois — APPROVED** [review]

**Review by sathish-soundarapandian — APPROVED** [review]

**Review by mankrit-singh — APPROVED** [review]

**Comment by vikyathharekal** [comment]
trigger review, uts

**Comment by vikyathharekal** [comment]
trigger ut5, ut6, ut7

**Comment by vikyathharekal** [comment]
trigger ut6

**Comment by vikyathharekal** [comment]
trigger uts

**Pull request merged**